src/main/java/core/basesyntax/service/ShopServiceImpl.java

```
@@ -0,0 +1,44 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final Map<String, Integer> stock = new HashMap<>();
+    private final OperationStrategy strategy;
+
+    public ShopServiceImpl(OperationStrategy strategy) {
+        if (strategy == null) {
+            throw new RuntimeException("OperationStrategy is null");
+        }
+        this.strategy = strategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        if (transactions == null) {
+            throw new RuntimeException("Transaction list is null");
+        }
+        for (FruitTransaction transaction : transactions) {
+            if (transaction == null) {
+                throw new RuntimeException("Transaction in the list is null");
+            }
+            OperationHandler handler = strategy.getHandler(transaction.getOperation());
```

`process(...)` retrieves the handler using `transaction.getOperation()` without checking whether the operation is null. The requirements state: "Handle null parameters across services and handlers with meaningful errors. [CHECKLIST ITEM #7]" Add an explicit null-check for `transaction.getOperation()` and throw a clear RuntimeException if it is null (for example: "Transaction operation is null for transaction: <transaction details>"). This ensures you fail fast with an informative message rather than passing a null into the strategy.